lib/egon/overcloud/undercloud_handle/node.rb

```diff
@@ -98,7 +96,11 @@ def introspect_node_status(node_uuid)
                          'X-Auth-Token' => auth_token},
             :method  => 'GET'
           })
-      Fog::JSON.decode(response.body)['finished']
+      finished = Fog::JSON.decode(response.body)['finished']
+      if finished
```

Comment:
you could just do `if Fog::JSON.decode(response.body)['finished']` but I like the readability.